code/modules/mob/living/silicon/robot/dogborg/dog_sleeper_vr.dm

```diff
@@ -431,7 +431,36 @@
 	name = "Brig-Belly"
 	desc = "Equipment for a K9 unit. A mounted portable-brig that holds criminals."
 	icon = 'icons/mob/dogborg_vr.dmi'
-	icon_state = "sleeper"
+	icon_state = "sleeperb"
 	inject_amount = 10
 	min_health = -100
 	injection_chems = null //So they don't have all the same chems as the medihound!
+
+/obj/item/device/dogborg/sleeper/compactor //Janihound gut.
+	name = "garbage processor"
+	desc = "A mounted garbage compactor unit with fuel processor."
+	icon = 'icons/mob/dogborg_vr.dmi'
+	icon_state = "compactor"
+	inject_amount = 10
+	min_health = -100
+	injection_chems = null //So they don't have all the same chems as the medihound!
+
+/obj/item/device/dogborg/sleeper/compactor/afterattack(var/obj/item/target, mob/living/silicon/user, proximity)
+	hound = loc
+	if(!proximity)
+		return
+	if(!isobj(target))
+		return
+	if(target.anchored)
+		return
+	if(target.w_class > ITEMSIZE_LARGE)
+		user << "<span class='warning'>This object is too large to fit into your [src.name]</span>"
```

Comment:
[src] was cramming in an extra "the" into the sentence which kinda felt grammatically off.